[DOC-2906] Added managed identity details for Azure Key Vault (#1479)
harness/developer-hub: docs/platform/Secrets/Secrets-Management/8-azure-key-vault.md

* Make sure you have set up an Azure account.
* Make sure you have **View** and **Create/Edit** permissions for secrets.

> ### Secret manager overview

Reviewer: It seems this doc doesn't include ## headings. It should start with with "## Before you begin". Some other heading levels below need to be adjusted accordingly.

Author: Done

> For a full overview of how your secrets are used with the Secrets Managers you configure in Harness, please see [Harness Secrets Management Overview](../Secrets-Management/1-harness-secret-manager-overview.md) and [Harness Security FAQs](../../../frequently-asked-questions/harness-faqs/harness-security-faqs.md).

Reviewer: ```suggestion
For a full overview of how your secrets are used with the Secrets Managers you configure in Harness, go to [Harness Secrets Management Overview](../Secrets-Management/1-harness-secret-manager-overview.md) and [Harness Security FAQs](../../../frequently-asked-questions/harness-faqs/harness-security-faqs.md).
```

Here's a visual summary:

![](../static/azure-key-vault-00.png)

### Limitations

> * Key Vault stores and manages secrets as sequences of octets (8-bit bytes), with a maximum size of 25k bytes each. For more information, see [Azure Key Vault secrets](https://docs.microsoft.com/en-us/azure/key-vault/secrets/about-secrets).

Reviewer: ```suggestion
* Key Vault stores and manages secrets as sequences of octets (8-bit bytes), with a maximum size of 25k bytes each. For more information, go to [Azure Key Vault secrets](https://docs.microsoft.com/en-us/azure/key-vault/secrets/about-secrets).
```

> ### Visual overview

Reviewer: Not sure what the purpose of this section is or what the image is. I would make a more specific heading, and introduce the image somehow.

Author: You are right. I removed it since it felt unnecessary.

:::

> #### PowerShell command

Reviewer: ```suggestion
### Create a Reader role using a PowerShell command
```

```
For details and examples, see Microsoft Azure's [Add or remove role assignments](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-powershell#application-at-a-subscription-scope) documentation.

> ### Add an Azure Key Vault secret manager in Harness

Author: Done

   * Copy the **Application (client) ID** for the Azure App registration you are using, and paste it into the Harness dialog's **Client ID** field.
   * Copy the **Directory (tenant) ID** of the Azure Active Directory (AAD) where you created your application, and paste it into the Harness dialog's **Tenant ID** field. (For details, see Microsoft Azure's [Get values for signing in](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-values-for-signing-in) topic.)
3. In **Subscription**, you can optionally enter your Azure Subscription ID (GUID).

   To find this ID, navigate to Azure's **Subscriptions** page, as outlined above in [Step 1: Create Azure Reader Role](../Secrets-Management/8-azure-key-vault.md#create-an-azure-reader-role). From the resulting list of subscriptions, copy the **Subscription ID** beside the subscription that contains your vaults.

   ![](../static/azure-key-vault-11.png)

   :::note
   If you do not enter a GUID, Harness uses the default subscription for the [Client ID](#setup-delegates) you've provided above.

>    :::

Reviewer: note is not rendering correctly, probably needs to be indented.

Author: Done

4. In **Key**, select **Create or Select a Secret**. For detailed steps on creating a new secret, go to [Add Text Secrets](../2-add-use-text-secrets.md).

   ![](../static/specify-credentials.png)

>    The secret that you reference here should have the Azure authentication key as the **Secret Value**. The below image shows the creation of a secret with Azure authentication key as its value:

Reviewer: ```suggestion
   The secret that you reference here should have the Azure authentication key as the **Secret Value**. The image below shows the creation of a secret with Azure authentication key as its value:
```

   ![](../static/azure-key-vault-13.png)

   To create and exchange the azure authentication key, perform the following steps:

>    * Navigate to Azure's **Certificates & secrets** page. (For details, see Microsoft Azure's [Create a new application secret](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-application-id-and-authentication-key) documentation.)

Author: Done.

   ![](../static/azure-key-vault-14.png)

   * Enter a **Description** and expiration option, then click **Add**.

   ![](../static/azure-key-vault-15.png)

   * Find your new key in the **Client secrets** section, and copy its value to your clipboard.

   ![](../static/azure-key-vault-16.png)

    
   :::note
   This is your only chance to view this key's value in Azure. Store the value somewhere secure, and keep it on your clipboard.Click **Continue**.
   :::

#### Use the credentials of a specific delegate

1. Select **Use the credentials of a specific Harness Delegate (IAM role, service account, managed identity, etc)**.
2. In **Subscription**, enter your Azure Subscription ID (GUID).
3. In **Environment**, select your environment.
4. In **Authentication**, select one of the following: 
   - **System Assigned Managed Identity**: If you select this, you need not provide any Ids.

     ![](../static/system-assigned-managed-identity.png)

   - **User Assigned Managed Identity**: If you select this, you need to provide the Application (client) Id in **Client Id**.
    
     ![](../static/user-assigned-managed-identity.png)

> ### Setup delegates

Reviewer: ```suggestion
### Set up delegates
```
